class-components/src/scripts/components/BottomSection.tsx

```diff
@@ -0,0 +1,131 @@
+import { Component } from 'react';
+import {
+  BottomSectionState,
+  SearchState,
+  Species,
+  Planets,
+  Vehicles,
+  Starships,
+  Films,
+  People,
+  Item,
+} from '../types/interfaces';
+
+class BottomSection extends Component<SearchState, BottomSectionState> {
+  constructor(props: SearchState) {
+    super(props);
+    this.state = {
+      items: [],
+      error: false,
+    };
+  }
+
+  componentDidMount() {
+    this.fetchItems(this.props.searchTerm);
+  }
+
+  componentDidUpdate(prevProps: SearchState) {
+    if (prevProps.searchTerm !== this.props.searchTerm) {
+      this.fetchItems(this.props.searchTerm);
+    }
+  }
+
+  fetchItems = async (searchTerm: string) => {
+    const endpoints = searchTerm
+      ? [
+          'https://swapi.dev/api/people/',
+          'https://swapi.dev/api/films/',
+          'https://swapi.dev/api/starships/',
+          'https://swapi.dev/api/vehicles/',
+          'https://swapi.dev/api/species/',
+          'https://swapi.dev/api/planets/',
+        ]
+      : ['https://swapi.dev/api/people/'];
+
+    try {
+      const fetches = endpoints.map((endpoint) => {
+        const url = searchTerm ? `${endpoint}?search=${searchTerm}` : endpoint;
+        return fetch(url).then((response) => response.json());
+      });
+
+      const results = await Promise.all(fetches);
```

Comment:
Have you considered using `Promise.allSettled()`? It  will take only one promise to be rejected for `Promise.all() `to be rejected while with `Promise.allSettled()` it's possible to process the successfully resolved promises if any